improver/utilities/temporal_interpolation.py

```diff
@@ -684,3 +684,249 @@ def process(self, cube_t0: Cube, cube_t1: Cube) -> CubeList:
+                fidelity period durations such that when the are summed up
+                they are equal to the original durations.
+        """
+        retotal = fidelity_period_cube.collapsed("time", iris.analysis.SUM)
+        factor = cube.data / retotal.data
+        # Masked points indicate divide by 0, set these points to 0. Also handle
+        # a case in which there is no masking on the factor array.
+        try:
+            factor = factor.filled(0)
+        except AttributeError:
+            factor[factor == np.inf] = 0
+
+        return factor
+
+    def construct_target_periods(self, fidelity_period_cube: Cube) -> Cube:
+        """Combine the short fidelity period cubes into cubes that describe
+        the target period.
+
+        Args:
+            fidelity_period_cube:
+                The short fidelity period cubes from which the target periods
+                are constructed.
+        Returns:
+            A cube containing the target period data with a time dimension
+            with an entry for each target period. These periods combined span
+            the original cube's period.
+        """
+        new_period_cubes = iris.cube.CubeList()
+
+        interval = timedelta(seconds=self.target_period)
+        start_time = fidelity_period_cube.coord("time").cell(0).bound[0]
+        end_time = fidelity_period_cube.coord("time").cell(-1).bound[-1]
+        while start_time < end_time:
+            period_constraint = iris.Constraint(
+                time=lambda cell: start_time <= cell.bound[0] < start_time + interval
+            )
+            components = fidelity_period_cube.extract(period_constraint)
+            component_cube = components.collapsed("time", iris.analysis.SUM)
+            component_cube.coord("time").points = component_cube.coord("time").bounds[
+                0
+            ][-1]
+            new_period_cubes.append(component_cube)
+            start_time += interval
+
+        return new_period_cubes.merge_cube()
+
+    def process(self, cube: Cube) -> Cube:
+        """Create target period duration diagnostics from the original duration
+        diagnostic data.
+
+        Args:
+            cube:
+                The original duration diagnostic cube.
+        Returns:
+            A cube containing the target period data with a time dimension
+            with an entry for each period. These periods combined span the
+            original cube's period.
+        Raises:
+            ValueError: The target period is not a factor of the input period.
+        """
+        period = self.cube_period(cube)
+        if period / self.target_period % 1 != 0:
+            raise ValueError(
+                "The target period must be a factor of the original period "
+                "of the input cube and the target period must <= the input "
+                "period. "
+                f"Input period: {period}, target period: {self.target_period}"
```

Comment:
I think that there's a "be" missing from this error message.

```suggestion
                "The target period must be a factor of the original period "
                "of the input cube and the target period must be <= the input "
                "period. "
                f"Input period: {period}, target period: {self.target_period}"
```